msteams-platform/messaging-extensions/api-based-secret-service-auth.md

```diff
@@ -0,0 +1,137 @@
+
+1. Select **Save**.
+
+The API key registration ID is update as the value for the `apiSecretRegistrationId` property in the app manifest.
+
+## Update app manifest
+
+Add an `apiSecretServiceAuthConfiguration` object with an `apiSecretRegistrationId` property, which contains the reference ID when you submit the API key through the Developer portal for Teams. For more information, see [composeExtensions.commands.](../resources/schema/manifest-schema.md#composeextensionscommands)
+
+```json
+"composeExtensions": [
+    {
+      "composeExtensionType": "apiBased",
+      "authorization": {
+        "authType": "apiSecretServiceAuth",
+        "apiSecretServiceAuthConfiguration": {
+            "apiSecretRegistrationId": "9xxxxb0f-xxxx-40cc-xxxx-15xxxxxxxxx3"
+        }
+      },
+```
+
+## Best practices
+
+* **Secret**:
+  * The secret value must have at least 10 characters and at most 128 characters.
+  * After you update the secret, it will take upto for one hour for the key to reflect throughout the system.
+
+* **Base URL**:
+  * The Base URL must begin with https, ensuring secure communication.
+  * Include the full host name to specify the exact domain.
+  * An optional path can be added to define a specific entry point for the API.
+
+   This structure is crucial for the security of your API secret(s), as Teams will only send secrets to endpoints that start with the specified Base URL.
+
+* **Target tenant**: As you develop your app within your Microsoft 365 tenant, you'll initially test it as a custom app built for your org (LoB) or custom app. During this stage, you must create the API secret registration with your **Home tenant** as the target tenant, ensuring the key remains exclusive to your tenant.
+
+  After you've completed testing and are ready to submit your app manifest to the Partner Center for the Teams Store, you'll need to switch the target tenant setting to **Any tenant**. This change allows your API secret registration to be used across various tenants once your app is available in the Teams Store.
+
+* **Teams app ID**: As you develop your app within your Microsoft 365 tenant and start to test it as a custom app built for your org (Lob) or custom app, you must set the API key registration with the Teams app ID as **Any Teams app**. This configuration allows the key to be used with any Teams app as custom app built for your org (Lob) or custom apps generate IDs after they're uploaded, and you won't have the app's ID at this stage.
+
+  Your key's security is still maintained through the **Home Tenant** and **Base URL**. When you're ready to release your app to the world, you need to change the Teams app ID setting to **Existing Teams app** and enter your manifest ID. Finally, submit your app manifest to the Partner Center for inclusion in the Teams Store. Later, your API secret registration is tied to your specific Teams app and can't be used with others.
```

Comment:
Change "manifest ID" to either "Teams app id"

Reply:
Updated as suggested